deepchem/feat/smiles_tokenizer.py

```diff
@@ -0,0 +1,357 @@
+        """
+
+        return self.ids_to_tokens.get(index, self.unk_token)
+
+    def convert_tokens_to_string(self, tokens):
+        """ Converts a sequence of tokens (string) in a single string.
+
+        Parameters
+        ----------
+        tokens: List[str]
+            List of tokens for a given string sequence.
+        """
+
+        out_string = " ".join(tokens).replace(" ##", "").strip()
+        return out_string
+
+    def add_special_tokens_ids_single_sequence(self, token_ids):
+        """
+        Adds special tokens to the a sequence for sequence classification tasks.
+        A BERT sequence has the following format: [CLS] X [SEP]
+
+        Parameters
+        ----------
+
+        token_ids: list[int]
+            list of tokenized input ids. Can be obtained using the encode or encode_plus methods.
+        """
+
+        return [self.cls_token_id] + token_ids + [self.sep_token_id]
+
+    def add_special_tokens_single_sequence(self, tokens):
+        """
+        Adds special tokens to the a sequence for sequence classification tasks.
+        A BERT sequence has the following format: [CLS] X [SEP]
+
+        Parameters
+        ----------
+        tokens: List[str]
+            List of tokens for a given string sequence.
+
+        """
+        return [self.cls_token] + tokens + [self.sep_token]
+
+    def add_special_tokens_sequence_pair(self, token_0: str, token_1: str) -> str:
+        """
+        Adds special tokens to a sequence pair for sequence classification tasks.
+        A BERT sequence pair has the following format: [CLS] A [SEP] B [SEP]
+
+        Parameters
+        ----------
+        token_0: str
+            The first token (A) in the sequence pair.
+        token_1: str
+            The second token (B) in the sequence pair.
+
+        Returns
+        -------
+        Sequence with added special tokens, [SEP] and [CLS], in the following format:
+        [CLS] A [SEP] B [SEP]
+
+        """
+
+        sep = [self.sep_token]
+        cls = [self.cls_token]
+
+        sequence_pair : str = cls + token_0 + sep + token_1 + sep
+
+        return sequence_pair
+
+    def add_special_tokens_ids_sequence_pair(self, token_ids_0 : List[int], token_ids_1: List[int]):
+        """
+        Adds special tokens to a sequence pair for sequence classification tasks.
+        A BERT sequence pair has the following format: [CLS] A [SEP] B [SEP]
+
+        Parameters
+        ----------
+        token_ids_0: List[int]
+            List of ids for the first string sequence in the sequence pair (A).
+
+        token_ids_1: List[int]
+            List of tokens for the second string sequence in the sequence pair (B).
+        """
+
+        sep = [self.sep_token_id]
+        cls = [self.cls_token_id]
+        return cls + token_ids_0 + sep + token_ids_1 + sep
+
+    def add_padding_tokens(self, token_ids, length, right=True):
```

Comment:
Type annotation

Reply:
added return type annotation to add_special_tokens_ids_sequence_pair()